source/geod24/concurrency.d

```diff
@@ -2212,3 +2215,111 @@ unittest
         });
     });
 }
+
+
+
+/***************************************************************************
+
+    Getter of Transceiver assigned to a called thread.
+
+    Returns:
+        Returns instance of `Transceiver` that is created by top thread.
+
+***************************************************************************/
+
+public @property Transceiver thisTransceiver () nothrow
+{
+    return thisInfo.transceiver;
+}
+
+
+/***************************************************************************
+
+    Setter of Transceiver assigned to a called thread.
+
+    Params:
+        value = The instance of `Transceiver`.
+
+***************************************************************************/
+
+public @property void thisTransceiver (Transceiver value) nothrow
+{
+    thisInfo.transceiver = value;
+}
+
+/*******************************************************************************
+
+    Transceiver device required for message exchange between threads.
+    Send and receive data requests, responses, commands, etc.
+
+*******************************************************************************/
+
+public interface Transceiver
+{
+    /***************************************************************************
+
+        It is a function that accepts Message
+
+        Params:
+            msg = The message to send.
+
+    ***************************************************************************/
+
+    void send (T) (T msg);
```

Comment:
Templated functions can't be virtual

Reply:
`Transceiver` was designed as interface, but is it a problem?
Structure Message is located in LocalRest.d, and Registry.d is referred to `Transceiver`.
I'd like to put this in LocalRest.d, but I couldn't.